libs/csm/rest/csm_rest_query_deployment.py

```diff
@@ -0,0 +1,218 @@
+
+    def __init__(self):
+        super(QueryDeployment, self).__init__()
+        self.local_sol_path = LOCAL_SOLUTION_PATH
+        self.prov_deploy_cfg = PROV_TEST_CFG["k8s_prov_cortx_deploy"]
+        self.rest_test_obj = RestTestLib()
+
+    def get_solution_yaml(self):
+        """
+        Read solution yaml into a dictionary
+        """
+        remote_sol_path = self.prov_deploy_cfg["git_remote_path"] + "solution.example.yaml"
+        self.log.info("Path for solution yaml on remote node: %s", remote_sol_path)
+        solution_path = self.rest_test_obj.master.copy_file_to_local(remote_path=remote_sol_path,
+                                                               local_path=self.local_sol_path)
+        data = yaml.safe_load(solution_path)
+        self.log.info("Printing solution yaml contents: %s", data)
+        return data
+
+    @RestTestLib.authenticate_and_login
+    def get_system_topology(self, uri_param=None, auth_header=None):
+        """
+        Get system topology
+        :param header: header for api authentication
+        :return: response
+        """
+        self.log.info("Printing auth header %s and query param %s ", auth_header, uri_param)
+        self.log.info("Get system topology request....")
+        if auth_header is None:
+            headers = self.headers
+        else:
+            headers = auth_header
+        endpoint = self.config["system_topology_endpoint"]
+        if uri_param is not None:
+            endpoint = endpoint + '/' + uri_param
+        self.log.info("system topology endpoint: %s", endpoint)
+        response = self.restapi.rest_call("get", endpoint=endpoint,
+                                          headers=headers)
+        self.log.info("Get system topology request successfully sent...")
+        return response
+
+    # Function not ready
+    def verify_system_topology(self, expected_response=HTTPStatus.OK):
+        """
+        Verify Get system topology
+        """
+        resp = self.get_system_topology()
+        result = True
+        if resp.status_code == expected_response:
+            result = True
+            get_response = resp.json()
+            self.log.info("Verify id fields have unique values")
+            self.log.info("Verfying cvg and devices in resp and solution yaml")
+            # Verify deployment version and deployment time
+        else:
+            self.log.error("Status code check failed.")
+            result = False
+        return result, get_response
+
+    @RestTestLib.authenticate_and_login
+    def get_cluster_topology(self, cluster_id=None, uri_param=None, auth_header=None):
+        """
+        Get cluster topology
+        :param header: header for api authentication
+        :cluster_id: id for the cluster
+        :return: response
+        """
+        self.log.info("Cluster id in cluster topology %s ", cluster_id)
+        self.log.info("Get cluster topology request....")
+        uri_param = 'clusters'
```

Comment:
uri_param is overwritten. 
u will need new variable name.

Reply:
changed to cluster_param